frontend/src/BodyPart.js

```diff
@@ -1,115 +1,80 @@
-import pic1 from './images/image-2.jpg';
-import pic2 from './images/image-3.jpg';
-import pic3 from './images/image-4.jpg';
-import pic4 from './images/image-1.jpg';
-import pic5 from './images/image-5.jpg';
-import pic6 from './images/image-6.jpg';
-import pic7 from './images/avocado.jpeg';
+import { getMostRecommendedRecipes } from './components/apiCalls';
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
 function BodyPart() {
+  const [recipeList, setRecipeList] = useState([]);
+  useEffect(() => {
+    const getInitData = async () => {
+      setRecipeList(await getMostRecommendedRecipes());
+    };
+    getInitData();
+  }, []);
+  console.log(recipeList);
   return (
     <>
       <div className="card-deck">
         <div className="card-deck" style={{ margin: '30px' }}>
-          <div className="card">
-            <img className="card-img-top" src={pic1} alt="pic1" />
-            <div className="card-body">
-              <h5 className="card-title">Italian Carbonara</h5>
-              <p className="card-text">
-                “Just a handful of ingredients makes a fantastic carbonara and,
-                done properly, it’s a thing of beauty. ”
-              </p>
-              <p className="card-text">
-                <small className="text-muted">Last updated 3 mins ago</small>
-              </p>
-            </div>
-          </div>
-          <div className="card">
-            <img className="card-img-top" src={pic2} alt="pic2" />
-            <div className="card-body">
-              <h5 className="card-title">Beef Stake with asparagus</h5>
-              <p className="card-text">
-                “If you’re cooking to impress, a juicy, flavour-packed steak
-                ticks a lot of boxes, and this dish takes your average steak
-                night totally different place."
-              </p>
-              <p className="card-text">
-                <small className="text-muted">Last updated 3 mins ago</small>
-              </p>
-            </div>
-          </div>
+          {recipeList
+            .map((recipe) => (
+              <Link
+                className="card"
+                style={{ width: '100%' }}
+                to={{
+                  pathname: `/recipes/${recipe.id}`,
+                  state: { ingredient: null },
+                }}
+              >
+                <img
+                  className="card-img-top"
+                  src={`/api/image/${recipe.image.id}`}
+                  alt={`${recipe.image.id}`}
+                  style={{ objectFit: 'cover', height: 550 }}
```

Comment:
Ha csak ebben a részben különbözik, akkor lehetne az index alapján elágazni: `index < 2 ? 550 : 217.8`
Ha több dologban akkor is lehet még finomítani a kódduplikáció helyett. Például lehet egy card komponens, aminek van egy highlighted propja, ami a különbségeket befolyásolja a flag alapján.